server/data/data-api/src/main/java/org/apache/james/user/api/model/User.java

```diff
@@ -34,13 +36,14 @@ public interface User {
     Username getUserName();
 
     /**
-     * Return true if pass matches password of this user.
-     * 
+     * Return translated username if the password is valid.
+     *
      * @param pass
      *            the password to test
-     * @return whether the password being tested is valid
+     * @return Optional of Username if the password is valid,
+     * otherwise return an empty Optional
      */
-    boolean verifyPassword(String pass);
+    Optional<Username> verifyPassword(String pass);
```

Comment:
I'm not sure we should modify this method.

Maybe getUsername needs to return the actual user value? And callers needs to adapt?

Reply:
> I'm not sure we should modify this method.

Do you mean only for the `User` scope or for `UsersRepository`  as well?

If you mean for `UsersRepository` as well, I am afraid that protocol callers could forget to get the translated username after authentication and we have bugs.